app/lib/shared/datastore_helper.dart

```diff
@@ -52,24 +52,29 @@ Future<T> withTransaction<T>(
   DatastoreDB db,
   Future<T> Function(TransactionWrapper tx) fn,
 ) async {
-  return db.withTransaction<T>((tx) async {
-    bool commitAttempted = false;
-    try {
-      final wrapper = TransactionWrapper._(tx);
-      final retval = await fn(wrapper);
-      if (wrapper._mutated) {
-        commitAttempted = true;
-        await tx.commit();
+  try {
+    return await db.withTransaction<T>((tx) async {
+      bool commitAttempted = false;
+      try {
+        final wrapper = TransactionWrapper._(tx);
+        final retval = await fn(wrapper);
+        if (wrapper._mutated) {
+          commitAttempted = true;
+          await tx.commit();
+        }
+        return retval;
+      } finally {
+        // If a commit has been attempted trying to rollback will always throw
+        // a StateError, hence, we never try to rollback if commit was attempted
+        if (!commitAttempted) {
+          await tx.rollback();
+        }
       }
-      return retval;
-    } finally {
-      // If a commit has been attempted trying to rollback will always throw
-      // a StateError, hence, we never try to rollback if commit was attempted
-      if (!commitAttempted) {
-        await tx.rollback();
-      }
-    }
-  });
+    });
+  } catch (e, st) {
+    _logger.info('Transaction failed.', e, st);
+    rethrow;
+  }
```

Comment:
Why do we need this? seems like noise...`withTransaction` is used by `withRetryTransaction` which has an `onRetry`.

We could question if we should even export `withTransaction` I'm not sure there is many places where you  shouldn't retry transactions. Maybe with likes it better to give up than to retry.. but probably not :D

Reply:
I though that we should have a generic logger around transactions and see a trace of what failed and where. (nothing else has changed in this method).

The visibility can be changed after we've migrated its uses to use `withRetryTransaction`.

Reply:
Can we make a version of `withTransaction` which doesn't do the logging, so that `withRetryTransaction` doesn't log entries it's retrying twice?